fhirstarter/functions.py

```diff
@@ -298,3 +446,34 @@ def _is_valid_parameter_name(name: str) -> bool:
         "resource",
         "type",
     }
+
+
+def _set_search_type_function_signature(
```

Comment:
The helper function that is used by both the async and non-async paths for search-type.